pkg/util/goroutinemap/goroutinemap_test.go

```diff
@@ -174,44 +275,73 @@ func Test_NewGoRoutineMap_Positive_ThirdOpAfterFirstCompletes(t *testing.T) {
 	}
 }
 
-func generateCallbackFunc(done chan<- interface{}) func() error {
-	return func() error {
-		done <- true
-		return nil
+func Test_NewGoRoutineMap_Positive_ThirdOpAfterFirstCompletesWithExpBackoff(t *testing.T) {
+	// Arrange
+	grm := NewGoRoutineMap(true /* exponentialBackOffOnError */)
+	operationName := "operation-name"
+	operation1DoneCh := make(chan interface{}, 0 /* bufferSize */)
+	operation1 := generateWaitFunc(operation1DoneCh)
+	err1 := grm.Run(operationName, operation1)
+	if err1 != nil {
+		t.Fatalf("NewGoRoutine failed. Expected: <no error> Actual: <%v>", err1)
 	}
-}
+	operation2 := generateNoopFunc()
+	operation3 := generateNoopFunc()
 
-func generateWaitFunc(done <-chan interface{}) func() error {
-	return func() error {
-		<-done
-		return nil
+	// Act
+	err2 := grm.Run(operationName, operation2)
+
+	// Assert
+	if err2 == nil {
+		t.Fatalf("NewGoRoutine did not fail. Expected: <Failed to create operation with name \"%s\". An operation with that name already exists.> Actual: <no error>", operationName)
+	}
+	if !IsAlreadyExists(err2) {
+		t.Fatalf("NewGoRoutine did not return alreadyExistsError, got: %v", err2)
 	}
-}
 
-func generatePanicFunc() func() error {
-	return func() error {
-		panic("testing panic")
+	// Act
+	operation1DoneCh <- true // Force operation1 to complete
+	err3 := retryWithExponentialBackOff(
+		time.Duration(20*time.Millisecond),
+		func() (bool, error) {
+			err := grm.Run(operationName, operation3)
+			if err != nil {
+				t.Logf("Warning: NewGoRoutine failed. Expected: <no error> Actual: <%v>. Will retry.", err)
```

Comment:
It is normal the run failed with error at the beginning, right? maybe move expected actual message, just say will retry?